---
title: >
  iSEEu_contribute_eurobioc2019
author:
- name: Federico Marini
  affiliation: 
  - &id1 Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz
  - Center for Thrombosis and Hemostasis (CTH), Mainz
  email: marinif@uni-mainz.de
- name: Charlotte Soneson
  affiliation: 
  - &id3 Friedrich Miescher Institute for Biomedical Research, Basel
  - SIB Swiss Institute of Bioinformatics
  email: charlottesoneson@gmail.com
- name: Kevin Rue-Albrecht
  affiliation: 
  - &id4 Kennedy Institute of Rheumatology, University of Oxford,
    Headington, Oxford OX3 7FY, UK.
  email: kevin.rue-albrecht@kennedy.ox.ac.uk
- name: Aaron Lun
  affiliation: 
  - &id2 Cancer Research UK Cambridge Institute, University of Cambridge
  email: infinite.monkeys.with.keyboards@gmail.com
date: "`r BiocStyle::doc_date()`"
package: "`r BiocStyle::pkg_ver('iSEE')`"
output: 
  BiocStyle::html_document:
    toc_float: true
vignette: >
  %\VignetteIndexEntry{iSEEu_contribute_eurobioc2019}
  %\VignetteEncoding{UTF-8}  
  %\VignettePackage{iSEEu}
  %\VignetteKeywords{GeneExpression, RNASeq, Sequencing, Visualization, QualityControl, GUI}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

**Compiled date**: `r Sys.Date()`

**Last edited**: 2019-11-27

**License**: `r packageDescription("iSEEu")[["License"]]`

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    error = FALSE,
    warning = FALSE,
    message = FALSE
)
stopifnot(requireNamespace("htmltools"))
htmltools::tagList(rmarkdown::html_dependency_font_awesome())
```

# Contributing to the iSEE universe (`iSEEu`)

The `iSEEu` package contains material and code that is useful to users of the `iSEE` package, but not strictly necessary for its function. 
We welcome contributions from the community, see below for more instructions. 

## Developer Day at the European Bioconductor 2019 conference 

The European Bioconductor 2019 conference (`#EuroBioc2019`) takes place at the UCLouvain, in Brussels (Belgium) - https://eurobioc2019.bioconductor.org.
During this conference, there will be a Developer Day event, where hackathon-like sessions will be focused around some topics proposed by the participants.

The `iSEE` development team proposed such a session, see here for more details: https://github.com/Bioconductor/EuroBioc2019/issues/4.
What follows is the original description of the topic, as a call to action for potential contributors to join in and design some "modes", i.e. preconfigured sets of panels and linked content to be used as starting setup when launching `iSEE`.
    
> The `iSEE` Bioconductor package (http://bioconductor.org/packages/iSEE/) enables interactive exploration of any type of data that can be stored in a `SummarizedExperiment` object. 
The default user interface of iSEE consists of a set of up to eight panels, each displaying a different aspect of the provided object (see e.g. http://shiny.imbei.uni-mainz.de:3838/iSEE/ for an example). 
The user is, however, free to change this interface interactively, by removing, adding or resizing existing panels, as well as by adding custom panels. 
In addition to changing the interface layout interactively, the user can pre-configure the set of panels, and even their content, and provide this information to `iSEE` when launching the application.
> 
> In an attempt to simplify this procedure for commonly used setups, we implemented a few different `iSEE` "modes" (https://github.com/csoneson/iSEE/blob/master/R/modes.R), each of which defines a set of panels and content for the resulting application.
> 
> The purpose of this "hackathon" session is to expand the set of `iSEE` modes to cover a wider range of common applications. 
The expected outcome is a collection of new modes suitable for various types of data analyses.
Depending on the participants, we will start by introducing the `iSEE` package and its capabilities.
> 
> Some suggestions for new modes to implement:
> 
> - scRNA-seq QC
> - Single panel type applications
> - bulk RNA-seq DE exploration

# Getting started

- install `iSEE` first - the development version is recommended.

```{r eval=FALSE}
BiocManager::install("iSEE", version = "devel")
# or
remotes::install_github("csoneson/iSEE")
```

- fork the `iSEEu` repo (https://github.com/csoneson/iSEEu) and clone it locally.

```
git clone https://github.com/[your_github_username]/iSEEu.git
```

- make the desired changes in the files - start from the `R` folder, then document via `roxygen2` - and push to your fork. 
  A typical contribution would be a function defining an `iSEE` mode, named `modeXXX`, where `XXX` provides a clear representation of the purpose of the mode. 
  Please place each mode in a script of its own, with the same name as the function. 
  The function should be documented (including an example), and any required packages should be added to the DESCRIPTION file. 
  Example data sets can often be obtained from an `ExperimentHub` package (e.g. from the [scRNAseq](http://bioconductor.org/packages/scRNAseq/) package for single-cell RNA-sequencing data), and should not be added to the `iSEEu` package.
  
- once your `mode` function is done, consider adding some information in the package.
  Some examples might be a screenshot of the mode in action (to be placed in the folder `inst/modes_img`), and a well-documented example use case (maybe an entry in the `vignettes` folder). Also add yourself as a contributor (`ctb`) to the DESCRIPTION file.

- make a pull request to the original repo - the GitHub site offers a practical framework to do so, enabling comments, code reviews, and other goodies.

- more on documenting and code guidelines:
  - if possible, please consider adding an example in the dedicated Roxygen preamble to show how to run each function
  - if possible, consider adding one or more unit tests - we use the `testthat` framework


# Some more info

#### Where do I look for constants within `iSEE`? {-}

Many of the "global" variables that are used in several places in `iSEE` are defined in the [constants.R](https://github.com/csoneson/iSEE/blob/master/R/constants.R) script in `iSEE`. 
We suggest to use these constants rather than hardcoding (e.g.) column names in the panel specification data frames, to protect against potential future changes of the precise column names. 
To access a constant, use `iSEE:::.constantName`. 

#### Is there any example I can check out to understand how things are supposed to work? {-}

There are several modes already defined in the `R/` directory. 

#### Are there any style guides I am supposed to follow? {-}

Yes. 
Mainly guided by common sense of "never changing a working system", please stick to the conventions we have been adopting for developing the existing codebase.
A few simple style options:

- keep the indentation as it is in the initial functions already available.
- if writing text (e.g. vignette), please use one sentence per line - this makes `git diff` operations easier to check.
- in code, use a degree of balance.
- for names, try to keep some consistency with what already is existing.
  We use camelCase for modes and some other functions, and prepend most unexported elements with a dot.

#### What if I need a custom panel type? {-}

In addition to the eight standard panel types, custom panels are easily accommodated within `iSEE` applications. 
For a guide, see the corresponding [vignette](https://bioconductor.org/packages/release/bioc/vignettes/iSEE/inst/doc/custom.html). 
For examples, see [this repo](https://github.com/kevinrue/iSEE_custom).

#### Where can I find a comprehensive introduction to `iSEE`? {-}

The `iSEE` package contains several vignettes detailing the main functionality. 
You can also take a look at this [workshop](https://kevinrue.github.io/iSEEWorkshop2019/index.html). 
A compiled version from the Bioc2019 conference (based on Bioconductor release 3.10) is available [here](http://biocworkshops2019.bioconductor.org.s3-website-us-east-1.amazonaws.com/page/iSEEWorkshop2019__iSEE-lab/).

