---
title: >
  iSEEu_contribute_eurobioc2019
author:
- name: Federico Marini
  affiliation: 
  - &id1 Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz
  - Center for Thrombosis and Hemostasis (CTH), Mainz
  email: marinif@uni-mainz.de
- name: Charlotte Soneson
  affiliation: 
  - &id3 Institute of Molecular Life Sciences, University of Zurich
  - SIB Swiss Institute of Bioinformatics
  email: charlottesoneson@gmail.com
- name: Kevin Rue-Albrecht
  affiliation: 
  - &id4 Kennedy Institute of Rheumatology, University of Oxford,
    Headington, Oxford OX3 7FY, UK.
  email: kevin.rue-albrecht@kennedy.ox.ac.uk
- name: Aaron Lun
  affiliation: 
  - &id2 Cancer Research UK Cambridge Institute, University of Cambridge
  email: infinite.monkeys.with.keyboards@gmail.com
date: "`r BiocStyle::doc_date()`"
package: "`r BiocStyle::pkg_ver('iSEE')`"
output: 
  BiocStyle::html_document:
    toc_float: true
vignette: >
  %\VignetteIndexEntry{iSEEu_contribute_eurobioc2019}
  %\VignetteEncoding{UTF-8}  
  %\VignettePackage{iSEEu}
  %\VignetteKeywords{GeneExpression, RNASeq, Sequencing, Visualization, QualityControl, GUI}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

**Compiled date**: `r Sys.Date()`

**Last edited**: 2019-11-27

**License**: `r packageDescription("iSEEu")[["License"]]`

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    error = FALSE,
    warning = FALSE,
    message = FALSE
)
stopifnot(requireNamespace("htmltools"))
htmltools::tagList(rmarkdown::html_dependency_font_awesome())
```

# Contributing to the iSEE universe (`iSEEu`)

[... some general info? ...Â ]

## Developer Day at the European Bioconductor 2019 conference 

The European Bioconductor 2019 conference (`#EuroBioc2019`) takes place at the UCLouvain, in Brussels (Belgium) - https://eurobioc2019.bioconductor.org.
During this conference, there will be a Developer Day event, where hackathon-like sessions will be focused around some topics proposed by the participants.

    The `iSEE` development team proposed such a session, see here for more details: https://github.com/Bioconductor/EuroBioc2019/issues/4.
    What follows is the original description of the topic, as a call to action for potential contributors to join in and design some "modes", i.e. preconfigured sets of panels and linked content to be used as starting setup when launching `iSEE`.
    
    The `iSEE` Bioconductor package (http://bioconductor.org/packages/iSEE/) enables interactive exploration of any type of data that can be stored in a `SummarizedExperiment` object. 
    The default user interface of iSEE consists of a set of up to eight panels, each displaying a different aspect of the provided object (see e.g. http://shiny.imbei.uni-mainz.de:3838/iSEE/ for an example). 
    The user is, however, free to change this interface interactively, by removing, adding or resizing existing panels, as well as by adding custom panels. 
    In addition to changing the interface layout interactively, the user can pre-configure the set of panels, and even their content, and provide this information to `iSEE` when launching the application.
    
    In an attempt to simplify this procedure for commonly used setups, we implemented a few different `iSEE` "modes" (https://github.com/csoneson/iSEE/blob/master/R/modes.R), each of which defines a set of panels and content for the resulting application.
    
    The purpose of this "hackathon" session is to expand the set of `iSEE` modes to cover a wider range of common applications. 
    The expected outcome is a collection of new modes suitable for various types of data analyses.
    Depending on the participants, we will start by introducing the `iSEE` package and its capabilities.
    
    Some suggestions for new modes to implement:
    
    - scRNA-seq QC
    - Single panel type applications
    - bulk RNA-seq DE exploration


## Getting started

- install `iSEE` first - the development version is recommended

```{r eval=FALSE}
BiocManager::install("iSEE", version = "devel")
# or
remotes::install_github("csoneson/iSEE")
```

- fork the `iSEEu` repo (https://github.com/csoneson/iSEEu) and clone it locally 

```
git clone https://github.com/[your_github_username]/iSEEu.git
```

- make the desired changes in the files - start from the `R` folder, then document via `roxygen2`

- make a pull request to the original repo - the GitHub site offers a practical framework to do so, enabling comments, code reviews, and other goodies

- once your `mode` object is done, consider adding some information in the package.
  Some examples might be a screenshot of the mode in action (to be placed in the folder `inst/modes_img`), and a well-documented example use case (maybe an entry in the `vignettes` folder)

## Some more info

#### Where do I look for constants within `iSEE`? {-}

...

#### Is there any example I can check out to understand how things are supposed to work? {-}

...

#### Are there any style guides I am supposed to follow? {-}

...


